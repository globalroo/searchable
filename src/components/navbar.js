import React from "react";
import { AppBar } from "@material-ui/core";
import { Link } from "@reach/router";
import { Toolbar } from "@material-ui/core";
import { Typography } from "@material-ui/core";
import Home from "@material-ui/icons/Home";

const styles = {
	appBarHeader: { background: "black" },
	titleLink: { color: "white", textDecoration: "none" },
	titleColor: { color: "white" },
	titleContainer: {
		display: "flex",
		flexDirection: "row",
		alignItems: "center"
	}
};

export const NavBar = ({ title }) => {
	return (
		<AppBar position="static" style={styles.appBarHeader}>
			<Toolbar>
				<Link style={styles.titleLink} to={`/`}>
					<div style={styles.titleContainer}>
						<Home />
						&nbsp;
						<Typography variant="title" style={styles.titleColor}>
							{title}
						</Typography>
					</div>
				</Link>
			</Toolbar>
		</AppBar>
	);
};
