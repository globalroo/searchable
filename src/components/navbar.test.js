import React from "react";
import { render, cleanup } from "react-testing-library";
import { NavBar } from "./navbar";
import "jest-dom/extend-expect";

describe("Asset Tile tests", () => {
	afterEach(cleanup);
	it("Renders a component with a navigate back wrapper", () => {
		const { container } = render(<NavBar title="Test Title" />);
		expect(container).toHaveTextContent("Test Title");
		expect(container).toMatchSnapshot();
	});
});
