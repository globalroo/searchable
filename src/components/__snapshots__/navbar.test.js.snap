// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Tile tests Renders a component with a navigate back wrapper 1`] = `
<div>
  <header
    class="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-8"
    style="background: black;"
  >
    <div
      class="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
    >
      <a
        aria-current="page"
        href="/"
        style="color: white; text-decoration: none;"
      >
        <div
          style="display: flex; flex-direction: row; align-items: center;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-41"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Â 
          <h2
            class="MuiTypography-root-50 MuiTypography-title-56"
            style="color: white;"
          >
            Test Title
          </h2>
        </div>
      </a>
    </div>
  </header>
</div>
`;
